app/Components/ChatApp.js

```diff
@@ -20,10 +35,27 @@ class ChatApp extends React.Component {
     this.props.socket.emit('sendchat', this.props.gameID, message);
   }
 
+  sendTypingState(typingState) {
+    this.props.socket.emit('sendtypingstate', this.props.gameID, this.props.user, typingState);
+  }
+
+  receiveTypingState(user, typingState) {
+    if (user != this.props.user)
+      this.setState({isTyping: typingState});
```

Comment:
This indicates whether the other user is typing, right?  I'd probably assume it's tracking the current user - a name like `isOtherUserTyping` might make that clearer.